### ISSUE TITLE
Get position of selected model?

### ISSUE THREAD

**Issue by antithing** [helix-toolkit/helix-toolkit#461]
Hi, Using the example `RectSelection` I have a 3d viewport in which I can select my objects, which are `BoxVisual3D`.

What I need to do, is return the 3d position of the selected object. I have:

```
  foreach (var model in models)
            {
                var geometryModel = model as GeometryModel3D;
                if (geometryModel != null)
                {
                    geometryModel.Material = geometryModel.BackMaterial = material;

                    
                    //do stuff 
                    UserControl1.Point1Position = model.Transform; 
                    
                    UserControl1.returnPoint.X = model.Transform.Value.M14;
                    UserControl1.returnPoint.Y = geometryModel.Transform.Value.M24;
                    UserControl1.returnPoint.Z = geometryModel.Transform.Value.M34;
                }
            }
```

But the values always return as `0`. (I spawn the box myself, so i know they are not `0`.
When I step through, there is a selected object, but the transform reads as all zeros. How can i get the positin of a `BoxVisual3D`?

Thanks.

**Comment by JeremyAnsel** [comment]
Hello,
You can try `model.Bounds.Location`.

**Comment by elijahFTL** [comment]
> JeremyAnsel commented 11 days ago
> Hello,
> You can try model.Bounds.Location.

This only returns the corner of the bounding object. It does not return the location the model was given in its transform. Attempting to use model.Bounds.Location and model.Bounds.Size together to find the centroid always results in (0, 0, 0)--this is because these coordinates are all local to the given model. I believe the question is asking how to get the model's global coordinates, and I came to this thread hoping to find out the same thing.

**Comment by antithing** [comment]
Thanks guys. I am using `model.Bounds.Location` for now, but I still need to find the centroid really.There MUST be a way to retrieve the Box coordinates!  Please post if you succeed.

**Comment by elijahFTL** [comment]
> Thanks guys. I am using model.Bounds.Location for now, but I still need to find the centroid really.There MUST be a way to retrieve the Box coordinates! Please post if you succeed.

You can find the centroid of the model using model.Bounds.

Let model.Bounds.Location = (x, y, z) and model.Bounds.Size = (a, b, c). The centroid is (x + a/2, y + b/2, z + c). However, you'll quickly notice that this always results in (0, 0, 0), because it's just local to the model itself. So it seems that there has to be another way out there.

**Comment by antithing** [comment]
There must be. It's crazy that you cannot retrieve the position of an object.

**Comment by elijahFTL** [comment]
I managed to solve the problem for my own very specific purpose. Each of my models corresponds to an instance of a class called Feature, which contains properties called Position (type Point3D) and Model (type GeometryModel3D). Feature.Position is used to transform the point when it is originally being placed. When I add a material to my model, I also make sure to store the geometry model as Model.

After this has been set up, if I hover the mouse over the model, I check for a collision. If the GeometryModel3D in the collision is EQUAL (using ==) to the Model property of one of my Feature instances (I loop through them all), I know it's a match and I can then just print out the coordinates from Feature.Position.

Long story short: I stored the true position and geometry model in the same object so that if a geometry model in the viewport is equal to it, I have a corresponding position in the same object.

**Comment by holance** [comment]
There is no "position" for a mesh, only triangle/vertices. If you do the model translation, you can get a translation vector. Or you have to create your own custom model with a "Position" property define by yourself.
